dot/network/service_test.go

```diff
@@ -44,12 +45,12 @@ var TestMessage = &BlockRequestMessage{
 var TestMessageTimeout = 3 * time.Second
 
 // helper method to create and start a new network service
-func createTestService(t *testing.T, cfg *Config) (node *Service, msgSend chan Message, msgRec chan Message) {
+func createTestService(t *testing.T, cfg *Config, blockState *MockBlockState) (node *Service, msgSend chan Message, msgRec chan Message) {
```

Comment:
as mentioned above, I think it would make more sense to not pass block state here but rather to create the mock block state within the `createTestService` method so that you do not have to duplicate it for every node and each node should have its own state, not share the same block state

Reply:
correct, however I need to have a reference to the blockstate for the testcase and have the blockstate being created pragmatically, not always the same with same dummy values. So passing it as a argument seemed to be the most reasonable option, given we have to customize it anyways.